src/sql/workbench/contrib/notebook/browser/notebookActions.ts

```diff
@@ -272,6 +271,102 @@ export class CollapseCellsAction extends ToggleableAction {
 	}
 }
 
+export class RunParametersAction extends TooltipFromLabelAction {
+	private static readonly label = localize('runParameters', "Run with Parameters");
+	private static readonly iconClass = 'icon-run-with-parameters';
+
+	constructor(id: string,
+		toggleTooltip: boolean,
+		context: URI,
+		@IQuickInputService private quickInputService: IQuickInputService,
+		@INotebookService private _notebookService: INotebookService,
+		@INotificationService private notificationService: INotificationService,
+	) {
+		super(id, {
+			label: RunParametersAction.label,
+			baseClass: baseIconClass,
+			iconClass: RunParametersAction.iconClass,
+			maskedIconClass: maskedIconClass,
+			shouldToggleTooltip: toggleTooltip
+		});
+	}
+
+	/**
+	 * Gets Default Parameters in Notebook from Parameter Cell
+	 * Uses that as Placeholder values for user to inject new values for
+	 * Once user enters all values it will open the new parameterized notebook
+	 * with injected parameters value from the QuickInput
+	*/
+	public async run(context: URI): Promise<void> {
+		const editor = this._notebookService.findNotebookEditor(context);
+		// Set defaultParameters to the parameter values in parameter cell
+		let defaultParameters = new Map<string, string>();
+		editor.cells.forEach(cell => {
+			if (cell.isParameter) {
+				for (let parameter of cell.source) {
+					let param = parameter.split('=', 2);
+					defaultParameters.set(param[0].trim(), param[1].trim());
+				}
+			}
+		});
+
+		// Store new parameters values the user inputs
+		let inputParameters = new Map<string, string>();
+		let uriParams = new URLSearchParams();
+		// Store new parameter values to map based off defaultParameters
+		if (defaultParameters.size === 0) {
+			// If there is no parameter cell indicate to user to create one
+			this.notificationService.notify({
+				severity: Severity.Info,
+				message: localize('noParametersCell', "This notebook cannot run with parameters until a parameter cell is added. [Learn more](https://docs.microsoft.com/sql/azure-data-studio/notebooks/notebooks-parameterization)."),
+			});
+			return;
+		} else {
+			for (let key of defaultParameters.keys()) {
+				let newParameterValue = await this.quickInputService.input({ prompt: key, value: defaultParameters.get(key) });
+				// If user cancels or escapes then it stops the action entirely
+				if (newParameterValue === undefined) {
+					return;
+				}
+				inputParameters.set(key, newParameterValue);
+			}
+			// Format the new parameters to be append to the URI
+			for (let key of inputParameters.keys()) {
+				// Will only add new injected parameters when the value is not the same as the defaultParameters values
+				if (inputParameters.get(key) !== defaultParameters.get(key)) {
+					// For empty strings we need to escape the value
+					// so that it is kept when adding uriParams.toString() to filePath
+					if (inputParameters.get(key) === '') {
```

Comment:
What about languages that don't support ' as their string delimiters?

Reply:
Currently, we are trying to make it generic but focused on Python kernel for parameterization. Most use cases for parameterization are through Python kernel. The other format would be supporting both ' and " as string delimiters.

Reply:
As long as that's expected. I will be curious to see how you go about solving this though in a generic way since there isn't really any way to tell how a language delimits its strings. Probably just have to go with a hardcoded mapping in the kernel specs or something I would imagine.